src/Persistence.php

```diff
@@ -435,30 +434,23 @@ protected function _typecastLoadField(Field $field, $value)
         // native DBAL DT types have no microseconds support
         if (in_array($field->type, ['datetime', 'date', 'time'], true)
             && str_starts_with(get_class($field->getTypeObject()), 'Doctrine\DBAL\Types\\')) {
-            if ($field->persist_format) {
-                $format = $field->persist_format;
-            } else {
-                // ! symbol in date format is essential here to remove time part of DateTime - don't remove, this is not a bug
-                $formats = ['date' => '+!Y-m-d', 'datetime' => '+!Y-m-d H:i:s', 'time' => '+!H:i:s'];
-                $format = $formats[$field->type];
-                if (str_contains($value, '.')) { // time possibly with microseconds, otherwise invalid format
-                    $format = preg_replace('~(?<=H:i:s)(?![. ]*u)~', '.u', $format);
-                }
+            $format = ['date' => 'Y-m-d', 'datetime' => 'Y-m-d H:i:s', 'time' => 'H:i:s'][$field->type];
+            if (str_contains($value, '.')) { // time possibly with microseconds, otherwise invalid format
+                $format = preg_replace('~(?<=H:i:s)(?![. ]*u)~', '.u', $format);
             }
 
-            if ($field->type === 'datetime') {
-                $value = \DateTime::createFromFormat($format, $value, new \DateTimeZone($field->persist_timezone));
-                if ($value !== false) {
+            $valueOrig = $value;
+            $value = \DateTime::createFromFormat('!' . $format, $value, new \DateTimeZone('UTC'));
+            if ($value !== false) {
+                if ($field->type === 'datetime') {
                     $value->setTimezone(new \DateTimeZone(date_default_timezone_get()));
                 }
```

Comment:
```suggestion
                } elseif ($field->type === 'date') {
                    // set time to 12:00 UTC so conversion to another TZ does not change the date
                    $value->setTime(12, 0);
                }
```

this can be interesting implementation, but it complicates assertions